<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>githubbeforesubmission</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__html"><p>Sara Saez-Atienzar, Sara Bandres-Ciga, Mike Nalls<br>
October 2019</p>
<h1 id="als_pathway-analysis-based-on-polygenic-risk-scores">ALS_Pathway analysis based on Polygenic Risk Scores</h1>
<h2 id="introduction">Introduction</h2>
<p>Despite the considerable progress made in unraveling the genetic causes of amyotrophic lateral sclerosis (ALS), we do not fully understand the molecular mechanisms underlying the disease. We analyzed genome-wide data involving nearly 80,000 individuals using a polygenic risk score approach to identify the biological pathways and cell types involved in ALS.</p>
<h2 id="overview">Overview</h2>
<p>We have performed a <strong>three-stage study design</strong> to identify pathways and cell types relevant to ALS risk</p>
<p><strong>Reference Dataset.</strong> A published GWAS study involving 12,577 ALS cases and 23,475 controls (<em>Van Rheenen et al., 2016</em>)<br>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/27455348">https://www.ncbi.nlm.nih.gov/pubmed/27455348</a></p>
<p><strong>Training Dataset</strong>. 	Plink Binary files containing individual-level data  genotype in our lab and  consisting in 5,653 ALS cases and 24,629 control subjects that were genotyped in our laboratory <em>(Nicolas et al., 2018)</em>.<br>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/29566793">https://www.ncbi.nlm.nih.gov/pubmed/29566793</a></p>
<p><strong>Replication Dataset.</strong> Plink Binary files containing individual-level data  genotype in our lab and  consisting in  2,420 ALS cases and 10,555 controls genotyped in our laboratory**(Nicolas et al., 2018)**<br>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/29566793">https://www.ncbi.nlm.nih.gov/pubmed/29566793</a></p>
<h1 id="pathway-analysis-based-on-polygenic-risk-scores-using-prsice-2">Pathway analysis based on Polygenic Risk Scores using PRSice-2</h1>
<p>A detailed information about PRSice2 can be found in the following link:<br>
<a href="https://choishingwan.github.io/PRSice/">https://choishingwan.github.io/PRSice/</a></p>
<h6 id="citation-choi-sw-and-o’reilly-pf.-prsice-2-polygenic-risk-score-software-for-biobank-scale-data.-gigascience-8-no.-7-july-1-2019.-httpsdoi.org10.1093gigasciencegiz082.">Citation: Choi SW, and O’Reilly PF. “PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.” GigaScience 8, no. 7 (July 1, 2019). <a href="https://doi.org/10.1093/gigascience/giz082">https://doi.org/10.1093/gigascience/giz082</a>.</h6>
<h2 id="requirements">Requirements</h2>
<h4 id="prsice2_linux-executable-v2.2.1">PRSice2_Linux executable (v2.2.1)</h4>
<h4 id="plink-v1.90b4.4-64-bit---www.cog-genomics.orgplink1.9">PLINK v1.90b4.4 64-bit   (<a href="http://www.cog-genomics.org/plink/1.9/">www.cog-genomics.org/plink/1.9/</a>)</h4>
<h4 id="plink-binary-files-with-pcas-and-a-covariate-file.-due-to-the-nature-of-prs-analysis-it-is-recommendable-to-include-20-pcs">PLINK binary files with PCAs and a covariate file. Due to the nature of PRS analysis, it is recommendable to include 20 PCs</h4>
<h4 id="r-version-3.2.3"><a href="https://www.r-project.org/">R</a> (<strong>version 3.2.3+</strong>)</h4>
<h4 id="prset-specific-input">PRSet Specific Input</h4>
<ul>
<li><strong>MSigDB file</strong>: File containing name of each gene sets and the ID of genes within the gene set on each individual line. If MSigDB is provided, GTF file is required.</li>
<li><strong>GTF file</strong>: A file contain the genome boundary of each individual gene</li>
</ul>
<h2 id="base-file">Base file</h2>
<p>Summary Stats without overlapping samples with training dataset. We have used a previous published GWAS summary Stats (Van Rheenen et al., 2016) <a href="https://www.ncbi.nlm.nih.gov/pubmed/27455348">https://www.ncbi.nlm.nih.gov/pubmed/27455348</a></p>
<h2 id="training-file">Training file</h2>
<p><strong>Training dataset</strong>: Raw genotype data of “target phenotype” in the form of PLINK binary<br>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/29566793">https://www.ncbi.nlm.nih.gov/pubmed/29566793</a></p>
<p><strong>Testing dataset</strong>: Raw genotype data of “target phenotype” in the form of PLINK binary<br>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/29566793">https://www.ncbi.nlm.nih.gov/pubmed/29566793</a></p>
<h1 id="code-for-training">CODE for training</h1>
<p>We first extracted the post clumping SNPs  (using default PRSice2 clumping options) from our training data set and rerun PRSice2 with the no-clump flag selected.</p>
<pre class=" language-bash"><code class="prism  language-bash">Rscript PRSice.R --cov-file ALL_COVS_ALS.txt --out Results -t Summary_Statistics_GWAS_2016 --beta --snp SNP --A1 A1 --A2 A2 --stat b --se se --pvalue p --ld workingData_BinaryPLINK --print-snp --score std --perm 1000 --prsice PRSice_linux -n 24 --binary-target T --maf 1E-2 --quantile 4 --prevalence 0.00005 --no-full --bar-levels 5E-2 --fastscore --gtf Homo_sapiens.GRCh37.75.gtf --msigdb c5.bp.v6.2.symbols.gmt.txt --multi-plot 10 --no-clump
</code></pre>
<h1 id="code-for-replication">CODE for Replication</h1>
<p>R</p>
<pre class=" language-bash"><code class="prism  language-bash">library<span class="token punctuation">(</span><span class="token string">"data.table"</span><span class="token punctuation">)</span>
VanRheenenTemp <span class="token operator">&lt;</span>- fread<span class="token punctuation">(</span><span class="token string">"Summary_Statistics_GWAS_2016"</span>, header <span class="token operator">=</span> T<span class="token punctuation">)</span>
temp <span class="token operator">&lt;</span>- VanRheenenTemp<span class="token punctuation">[</span>order<span class="token punctuation">(</span>VanRheenenTemp<span class="token variable">$SNP</span>,VanRheenenTemp<span class="token variable">$p</span><span class="token punctuation">)</span>,<span class="token punctuation">]</span>
temp<span class="token variable">$dupe</span> <span class="token operator">&lt;</span>- duplicated<span class="token punctuation">(</span>temp<span class="token variable">$SNP</span><span class="token punctuation">)</span>
VanRheenen <span class="token operator">&lt;</span>- subset<span class="token punctuation">(</span>temp, dupe <span class="token operator">==</span> F<span class="token punctuation">)</span>
listOfPathways <span class="token operator">&lt;</span>- read.table<span class="token punctuation">(</span><span class="token string">"significant.txt"</span>, header <span class="token operator">=</span> T<span class="token punctuation">)</span>
names<span class="token punctuation">(</span>listOfPathways<span class="token punctuation">)</span> <span class="token operator">&lt;</span>- c<span class="token punctuation">(</span><span class="token string">"id"</span><span class="token punctuation">)</span>
for<span class="token punctuation">(</span>i <span class="token keyword">in</span> 1:length<span class="token punctuation">(</span>listOfPathways<span class="token variable">$id</span><span class="token punctuation">))</span>
<span class="token punctuation">{</span>
	instrumentId <span class="token operator">&lt;</span>- as.character<span class="token punctuation">(</span>listOfPathways<span class="token variable">$id</span><span class="token punctuation">[</span>i<span class="token punctuation">]</span><span class="token punctuation">)</span>
	thisDataTemp <span class="token operator">&lt;</span>- fread<span class="token punctuation">(</span>file <span class="token operator">=</span> instrumentId, header <span class="token operator">=</span> F<span class="token punctuation">)</span>
	thisData <span class="token operator">&lt;</span>- subset<span class="token punctuation">(</span>thisDataTemp, V5 <span class="token operator">==</span><span class="token string">"Y"</span><span class="token punctuation">)</span>
	merged <span class="token operator">&lt;</span>- merge<span class="token punctuation">(</span>thisData, VanRheenen, by.x <span class="token operator">=</span> <span class="token string">"V1"</span>, by.y<span class="token operator">=</span><span class="token string">"SNP"</span><span class="token punctuation">)</span>
	merged<span class="token variable">$allele</span> <span class="token operator">&lt;</span>- toupper<span class="token punctuation">(</span>merged<span class="token variable">$A1</span><span class="token punctuation">)</span>
	toPlink <span class="token operator">&lt;</span>- merged<span class="token punctuation">[</span>,c<span class="token punctuation">(</span><span class="token string">"V1"</span>,<span class="token string">"allele"</span>,<span class="token string">"b"</span><span class="token punctuation">)</span><span class="token punctuation">]</span>
	write.table<span class="token punctuation">(</span>toPlink, <span class="token function">file</span> <span class="token operator">=</span> paste<span class="token punctuation">(</span>instrumentId,<span class="token string">".significant.txt"</span>,sep <span class="token operator">=</span> <span class="token string">""</span><span class="token punctuation">)</span>, quote <span class="token operator">=</span> F, sep <span class="token operator">=</span> <span class="token string">" "</span>, col.names <span class="token operator">=</span> F, row.names <span class="token operator">=</span> F<span class="token punctuation">)</span>
<span class="token punctuation">}</span>	
</code></pre>
<p>PLINK</p>
<pre class=" language-bash"><code class="prism  language-bash"><span class="token function">cat</span> significant.txt <span class="token operator">|</span> <span class="token keyword">while</span> <span class="token function">read</span> LINE 
<span class="token keyword">do</span>
<span class="token keyword">echo</span> <span class="token string">"THIS IS LINE"</span> <span class="token variable">$LINE</span>
plink --bfile REPLICATION --score <span class="token variable">$LINE</span>.bpe.txt --out <span class="token variable">$LINE</span>.significanttoreplication
<span class="token keyword">done</span>	
</code></pre>
<p>R</p>
<pre class=" language-bash"><code class="prism  language-bash">library<span class="token punctuation">(</span><span class="token string">"data.table"</span><span class="token punctuation">)</span>
listOfProfiles <span class="token operator">&lt;</span>- read.table<span class="token punctuation">(</span><span class="token string">"significant.txt"</span>, header <span class="token operator">=</span> T<span class="token punctuation">)</span>
names<span class="token punctuation">(</span>listOfProfiles<span class="token punctuation">)</span> <span class="token operator">&lt;</span>- c<span class="token punctuation">(</span><span class="token string">"id"</span><span class="token punctuation">)</span>
covs1 <span class="token operator">&lt;</span>- fread<span class="token punctuation">(</span><span class="token string">"/data/LNG/ALS_PATHWAYS/PHENO_COVARIATES_ALS_LNGonly_unrelated.txt"</span>, header <span class="token operator">=</span> T<span class="token punctuation">)</span>
covs2 <span class="token operator">&lt;</span>- fread<span class="token punctuation">(</span><span class="token string">"/data/LNG/ALS_PATHWAYS/ALS_Hits_toCOVs.raw"</span>, header <span class="token operator">=</span> T<span class="token punctuation">)</span>
covs <span class="token operator">&lt;</span>- merge <span class="token punctuation">(</span>covs1, covs2, by <span class="token operator">=</span><span class="token string">"FID"</span><span class="token punctuation">)</span>
covs<span class="token variable">$index</span> <span class="token operator">&lt;</span>- paste<span class="token punctuation">(</span>covs<span class="token variable">$FID</span>,covs<span class="token variable">$FID</span>, sep <span class="token operator">=</span> <span class="token string">"_"</span><span class="token punctuation">)</span>
covs<span class="token variable">$CASE</span> <span class="token operator">&lt;</span>- covs<span class="token variable">$PHENO</span> - 1
outPut <span class="token operator">&lt;</span>- matrix<span class="token punctuation">(</span>ncol <span class="token operator">=</span> 4, nrow <span class="token operator">=</span> length<span class="token punctuation">(</span>listOfProfiles<span class="token variable">$id</span><span class="token punctuation">)</span>, NA<span class="token punctuation">)</span>
colnames<span class="token punctuation">(</span>outPut<span class="token punctuation">)</span> <span class="token operator">&lt;</span>- c<span class="token punctuation">(</span><span class="token string">"pathway"</span>,<span class="token string">"b"</span>,<span class="token string">"se"</span>,<span class="token string">"p"</span><span class="token punctuation">)</span>
for<span class="token punctuation">(</span>i <span class="token keyword">in</span> 1:length<span class="token punctuation">(</span>listOfProfiles<span class="token variable">$id</span><span class="token punctuation">))</span>
<span class="token punctuation">{</span>
	profileName <span class="token operator">&lt;</span>- 	as.character<span class="token punctuation">(</span>listOfProfiles<span class="token variable">$id</span><span class="token punctuation">[</span>i<span class="token punctuation">]</span><span class="token punctuation">)</span>
	profile <span class="token operator">&lt;</span>- fread<span class="token punctuation">(</span>file <span class="token operator">=</span> paste<span class="token punctuation">(</span>profileName, <span class="token string">".significanttoreplication.profile"</span>, sep <span class="token operator">=</span> <span class="token string">""</span><span class="token punctuation">)</span>, header <span class="token operator">=</span> T<span class="token punctuation">)</span>
	profile<span class="token variable">$index</span> <span class="token operator">&lt;</span>- paste<span class="token punctuation">(</span>profile<span class="token variable">$FID</span>, profile<span class="token variable">$IID</span>, sep <span class="token operator">=</span> <span class="token string">"_"</span><span class="token punctuation">)</span>
	data <span class="token operator">&lt;</span>- merge<span class="token punctuation">(</span>covs, profile, by <span class="token operator">=</span> <span class="token string">"index"</span><span class="token punctuation">)</span>
	meanControls <span class="token operator">&lt;</span>- mean<span class="token punctuation">(</span>data<span class="token variable">$SCORE</span><span class="token punctuation">[</span>data<span class="token variable">$CASE</span> <span class="token operator">==</span> 0<span class="token punctuation">]</span><span class="token punctuation">)</span>
	sdControls <span class="token operator">&lt;</span>- sd<span class="token punctuation">(</span>data<span class="token variable">$SCORE</span><span class="token punctuation">[</span>data<span class="token variable">$CASE</span> <span class="token operator">==</span> 0<span class="token punctuation">]</span><span class="token punctuation">)</span>
	data<span class="token variable">$zSCORE</span> <span class="token operator">&lt;</span>- <span class="token punctuation">(</span>data<span class="token variable">$SCORE</span> - meanControls<span class="token punctuation">)</span>/sdControls	
	grsTest <span class="token operator">&lt;</span>- glm<span class="token punctuation">(</span>CASE ~ zSCORE + age_at_onset + gender + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10 + PC11 + PC12 + PC13 + PC14 + PC15 + PC16 + PC17 + PC18 + PC19 + PC20, family<span class="token operator">=</span><span class="token string">"binomial"</span>, data <span class="token operator">=</span> data<span class="token punctuation">)</span>
	beta <span class="token operator">&lt;</span>- summary<span class="token punctuation">(</span>grsTest<span class="token punctuation">)</span><span class="token variable">$coefficients</span><span class="token punctuation">[</span><span class="token string">"zSCORE"</span>,<span class="token string">"Estimate"</span><span class="token punctuation">]</span>
	se <span class="token operator">&lt;</span>- summary<span class="token punctuation">(</span>grsTest<span class="token punctuation">)</span><span class="token variable">$coefficients</span><span class="token punctuation">[</span><span class="token string">"zSCORE"</span>,<span class="token string">"Std. Error"</span><span class="token punctuation">]</span>
	p <span class="token operator">&lt;</span>- summary<span class="token punctuation">(</span>grsTest<span class="token punctuation">)</span><span class="token variable">$coefficients</span><span class="token punctuation">[</span><span class="token string">"zSCORE"</span>,<span class="token string">"Pr(&gt;|z|)"</span><span class="token punctuation">]</span>
	outPut<span class="token punctuation">[</span>i,1<span class="token punctuation">]</span> <span class="token operator">&lt;</span>- profileName
	outPut<span class="token punctuation">[</span>i,2<span class="token punctuation">]</span> <span class="token operator">&lt;</span>- beta
	outPut<span class="token punctuation">[</span>i,3<span class="token punctuation">]</span> <span class="token operator">&lt;</span>- se
	outPut<span class="token punctuation">[</span>i,4<span class="token punctuation">]</span> <span class="token operator">&lt;</span>- p
<span class="token punctuation">}</span>
write.table<span class="token punctuation">(</span>outPut, <span class="token string">"REPLICATION_plink.tab"</span>, quote <span class="token operator">=</span> F, sep <span class="token operator">=</span> <span class="token string">"\t"</span>, row.names <span class="token operator">=</span> F<span class="token punctuation">)</span>
</code></pre>
</div>
</body>

</html>
